dataplane/linux/int_dataplane.go

```diff
@@ -866,6 +886,75 @@ func (d *InternalDataplane) RecvMessage() (interface{}, error) {
 	return <-d.fromDataplane, nil
 }
 
+// configureMTU auto-detects the MTU to use for IPIP, VXLAN, and Wireguard interfaces
+// if not already set, and writes the smallest MTU to disk for use by other components.
+func configureMTU(config *Config) error {
```

Comment:
Recommend splitting this into a detection function and a function to write out the data.  Then, you can periodically call the detection function in a loop and, if the MTU changes, trigger felix to restart so that we apply the new MTU.